mobile/src/main/java/ch/epfl/sdp/mobile/ui/social/SwipeToUnfollow.kt

```diff
@@ -0,0 +1,106 @@
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.gestures.Orientation.Horizontal
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.round
+import androidx.compose.ui.zIndex
+import ch.epfl.sdp.mobile.state.LocalLocalizedStrings
+import ch.epfl.sdp.mobile.ui.Delete
+import ch.epfl.sdp.mobile.ui.PawniesIcons
+
+/** The possible states for a [SwipeToUnfollow] composable. */
+private enum class SwipeToUnfollow {
+
+  /** The action is hidden. */
+  Closed,
+
+  /** The action is visible. */
+  Open,
+}
+
+/**
+ * Displays the [content] with a swipe-to-unfollow action.
+ *
+ * @param onUnfollowClick the callback which is called when the unfollow action is performed.
+ * @param modifier the [Modifier] for this composable.
+ * @param enabled true iff the swipe-to-delete action should be enabled.
+ * @param content the body to display in the composable.
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun SwipeToUnfollow(
+    onUnfollowClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    content: @Composable () -> Unit,
+) {
+  val state = rememberSwipeableState(SwipeToUnfollow.Closed)
+
+  val anchors =
+      mapOf(
+          0f to SwipeToUnfollow.Closed,
+          with(LocalDensity.current) { 72.dp.toPx() } to SwipeToUnfollow.Open,
+      )
+
+  Box(
+      modifier
+          .height(IntrinsicSize.Min)
+          .swipeable(state = state, anchors = anchors, orientation = Horizontal)) {
+    val isSwipeInProgress by remember {
+      derivedStateOf {
+        state.isAnimationRunning || state.progress.fraction != 1f || state.overflow.value != 0f
+      }
+    }
+    UnfollowBackground(
+        onClick = onUnfollowClick,
+        modifier =
+            Modifier.width(72.dp)
```

Comment:
Nitpick : extract this `72.dp` to a constant, since it's used in multiple places.

Reply:
resolved